hc/front/tests/test_add_channel.py

```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
             self.assertContains(r, "Integration Settings", status_code=200)
 
     ### Test that the team access works
+    def test_teamaccess_works(self):
+        url = "/integrations/add/"
+        form = {"kind": "email", "value": "alice@example.com"}
+
+        self.client.login(username="alice@example.org", password="password")
+        self.client.post(url, form)
+        alice_channels = Channel.objects.filter(user=self.alice)
+        self.client.logout()
+
+        self.client.login(username="bob@example.org", password="password")
+        r = self.client.get("/integrations/")
+        self.assertContains(r, alice_channels.first().code)
+        self.assertEquals(r.status_code, 200)
+
+
+
     ### Test that bad kinds don't work
+    def test_bad_kinds_dont_work(self):
+        # Bad kinds means that a user can't add unsupported intergrations.
+        self.client.login(username="alice@example.org")
+        bad_kinds = ("whatsapp", "facebook", "linkedin")
```

Comment:
It is not necessary to loop in tests.
Just use one bad kind and test against it.

Reply:
Noted.